ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,211 @@
+    examples = """
+    To merge DL1-files created by the stage1-process-tool from a specific directory:
+    > ctapipe-merge-dl1 --input=/path/to/directory --ouput=/path/to/output_file.h5 --progress
+    """
+    input_dir = Path(help="input dl1-directory",
+                     exists=None, directory_ok=True, file_ok=False).tag(config=True)
+    output_path = Path(help="Merged-DL1 output filename").tag(config=True)
+    skip_images = Bool(help="Skip DL1/Event/Image data in output",
+                       default_value=False).tag(config=True)
+    skip_parameters = Bool(help="Skip image parameters",
+                           default_value=False).tag(config=True)
+    overwrite = Bool(help="overwrite output file if it exists").tag(config=True)
+    progress_bar = Bool(help="show progress bar during processing").tag(config=True)
+
+    aliases = {'input': 'DL1MergeTool.input_dir',
+               'output': 'DL1MergeTool.output_path'}
+
+    flags = {
+        "skip-images": (
+            {"DL1MergeTool": {"skip_images": True}},
+            "store DL1/Event/Telescope images in output",
+        ),
+        "skip-parameters": (
+            {"DL1MergeTool": {"skip_parameters": True}},
+            "store DL1/Event/Telescope parameters in output",
+        ),
+        "overwrite": (
+            {"DL1MergeTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"DL1MergeTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        )
+    }
+
+    def setup(self):
+        # prepare output path:
+        self.output_path = self.output_path.expanduser()
+        if self.output_path.exists():
+            if self.overwrite:
+                self.log.warning(f"Overwriting {self.output_path}")
+                self.output_path.unlink()
+            else:
+                self.log.critical(f"Output file {self.output_path} exists, "
+                                   "use `--overwrite` to overwrite")
+                sys.exit(1)
+
+        if self.skip_parameters is True and self.skip_images is True:
+            self.log.warning('Skip-parameters and skip-images are both'
+                             'set to True')
+
+        self.input_path_list = sorted(self.input_dir.iterdir())
+
+        # create output file with subarray from first file
+        self.first_subarray = SubarrayDescription.from_hdf(self.input_path_list[0])
+        self.first_subarray.to_hdf(self.output_path)
+        self.output_file = tables.open_file(self.output_path, mode='a')
+
+    def add_image_statistics(self, file):
+        # Creates table for image statistics and adds the entries together.
+        # This does not append rows to the existing table
+        if table_path_image_statistics not in self.output_file:
+            head, tail = os.path.split(os.path.split(table_path_image_statistics)[0])
```

Comment:
I suggest also to use `pathlib` instead of os.path everywhere.  It's much cleaner

Reply:
This is a path inside a hdf5 file, so I think just using string operation is better.

Reply:
I needed it twice, because there was the node at the end of this path, so i had to split it once to get the path from the node as 'head' and twice to create the group, because i need the end from 'head' of first split. But i changed it anyway